data-prepper-plugins/common/src/main/java/org/opensearch/dataprepper/common/utils/RetryUtil.java

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.dataprepper.common.utils;
+
+public class RetryUtil {
+    private static final int MAX_RETRIES = 3;
+    private static final long BASE_DELAY_MS = 100; // 100ms base delay for exponential backoff
+
+    public static boolean retryWithBackoff(Runnable task) {
+        int attempt = 0;
+        while (attempt < MAX_RETRIES) {
+            try {
+                task.run();
+                return true; // Success
+            } catch (Exception e) {
+                try {
+                    Thread.sleep(BASE_DELAY_MS * (1L << attempt)); // Exponential backoff
```

Comment:
This will increase quite fast with number of retries. We should use `import com.linecorp.armeria.client.retry.Backoff;` and the code should be similar to exponential backoff code in `https://github.com/kkondaka/kk-data-prepper-f2/blob/main/data-prepper-plugins/opensearch/src/main/java/org/opensearch/dataprepper/plugins/sink/opensearch/BulkRetryStrategy.java`